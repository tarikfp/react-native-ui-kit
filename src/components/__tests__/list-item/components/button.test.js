import * as React from "react";
import renderer from "react-test-renderer";
import ListItemButton from "../../../list-item/components/button";

describe("tests list item with button component", () => {
  it("renders list item button with alignment", () => {
    const tree = renderer
      .create(<ListItemButton label="test" alignment="end" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item button with custom wrapperStyle", () => {
    const tree = renderer
      .create(
        <ListItemButton label="test" wrapperStyle={{ marginBottom: 40 }} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
