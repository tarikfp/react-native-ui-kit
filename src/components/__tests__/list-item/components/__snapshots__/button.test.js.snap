// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests list item with button component renders list item button with alignment 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderColor": "rgba(0, 0, 0,0.05)",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 35,
        "justifyContent": "center",
        "minWidth": 83,
        "opacity": 1,
        "padding": 8,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": undefined,
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 18,
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;

exports[`tests list item with button component renders list item button with custom wrapperStyle 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "marginBottom": 40,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderColor": "rgba(0, 0, 0,0.05)",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 35,
        "justifyContent": "center",
        "minWidth": 83,
        "opacity": 1,
        "padding": 8,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": undefined,
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 18,
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
